### PR TITLE
[SEARCH-396] Use warnings instead of errors for search-alias views

### PR DESCRIPTION
**Pull request by gnusi**
### Scope & Purpose

Use warnings instead of errors for search-alias views

- [x] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.10: *(Please link PR)*
  - [ ] Backport for 3.9: *(Please link PR)*
  - [ ] Backport for 3.8: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR: https://github.com/arangodb/enterprise/pull/1080
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

**Review by MBkkt — APPROVED** [review]

File: arangod/IResearch/IResearchFilterFactory.cpp [hunk 1]
```diff
@@ -104,25 +104,33 @@ Result fromFuncMinHashMatch(char const* funcName, irs::boolean_filter* filter,
...
+    if (ctx) {
+      ctx->registerWarning(TRI_ERROR_BAD_PARAMETER, error);
+    }
+  };
+
   auto const& analyzer = (*fieldAnalyzerProvider)(name);
+  if (!analyzer) {
+    // Only possible with SearchAlias views
+    registerWarning(absl::StrCat("Analyzer for field '", name, "' isn't set"));
+    return FieldMeta::identity();
```
Comment (MBkkt) [inline arangod/IResearch/IResearchFilterFactory.cpp:125]: contextAnalyzer ? contextAnalyzer : FieldMeta::identity();

Comment (Dronplane) [reply arangod/IResearch/IResearchFilterFactory.cpp:125]: contextAnalyzer  looks more consistent in this case so +1 from me

**Review by Dronplane — APPROVED** [review]
LGTM

File: arangod/IResearch/IResearchFilterFactory.cpp [hunk 2]
```diff
@@ -104,27 +104,11 @@ Result fromFuncMinHashMatch(char const* funcName, irs::boolean_filter* filter,
...
-                      "' doesn't match field '", name, "' analyzer '",
-                      analyzer ? analyzer->name() : "", "'")};
-  }
-  if (!analyzer) {
-    r = {TRI_ERROR_BAD_PARAMETER,
-         absl::StrCat("Analyzer for field '", name, "' isn't set")};
-  }
-  return analyzer;
+    std::string_view name, aql::ExpressionContext* ctx) const noexcept {
+  // FIXME(gnusi): refactor this function
```
Comment (MBkkt) [inline arangod/IResearch/IResearchFilterFactory.cpp:108]:
```suggestion
```

File: arangod/IResearch/Search.cpp [hunk 3]
```diff
@@ -322,18 +323,42 @@ AnalyzerProvider SearchMeta::createProvider(
...
-    if (it != analyzers.end() && it->second.includeAllFields) {
-      return it->second.analyzer;
+
+    if (it == analyzers.end()) {
+      registerWarning(
+          absl::StrCat("Analyzer for field '", field, "' isn't set"));
+      return contextAnalyzer ? contextAnalyzer : FieldMeta::identity();
+    }
+
+    auto& analyzer = it->second.analyzer;
```
Comment (MBkkt) [inline arangod/IResearch/Search.cpp:352]: getAnalyzer(above) can return null

**Comment by gnusi** [comment]
https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/24342/

**Pull request merged**